index.js

```diff
@@ -68,28 +112,14 @@ class ExecutorRouter extends Executor {
      * @return {Promise}
      */
     _start(config) {
-        const annotations = config.annotations || {};
-        const executorType = annotations[ANNOTATION_EXECUTOR_TYPE];
-        const executor = this[executorType] || this.defaultExecutor; // Route to executor (based on annotations) or use default executor
-
-        return executor.start(config);
-    }
+        const options = this.getFullConfig(config);
```

Comment:
Agree with Darren. Can wrap getFullConfig as a promise, and call it before calling this.verifyConnection.
example: https://github.com/screwdriver-cd/datastore-base/blob/master/index.js#L15